### PR TITLE
Optimize Instructions cache

### PR DESCRIPTION
**Pull request by shargon**
Use Script class for caching the instructions instead of ExecutionContext

### CODE REVIEW

**Comment by shargon** [comment]
Previously we cached the instructions inside ExecutionContext, so if you made a CALL inside other call, you will create two ExecutionContexts with different caches, after this change, the script have this cache, so every ExecutionContext who comes from one Script, share this cache.

**Comment by igormcoelho** [comment]
As soon as this is merged, we can improve https://github.com/neo-project/neo-vm/pull/121 with the final tests missing here.

File: src/neo-vm/ExecutionContext.cs [hunk 1]
```diff
@@ -1,16 +1,13 @@
...
 {
     public class ExecutionContext
     {
-        private readonly Dictionary<int, Instruction> instructions = new Dictionary<int, Instruction>();
-
         /// <summary>
         /// Number of items to be returned
         /// </summary>
-        internal int RVCount { get; }
+        public int RVCount { get; }
```
Comment (erikzhang) [inline src/neo-vm/ExecutionContext.cs]: Why change this to `public`?

File: src/neo-vm/Script.cs [hunk 2]
```diff
@@ -8,6 +9,7 @@ public class Script
 
         private readonly byte[] _value;
         private readonly ICrypto _crypto;
+        private readonly IDictionary<int, Instruction> _instructions = new Dictionary<int, Instruction>();
```
Comment (erikzhang) [inline src/neo-vm/Script.cs]: I think it's not necessary to use interface for a `private` field.

File: src/neo-vm/ExecutionContext.cs (hunk 1 above)
Comment (shargon) [reply src/neo-vm/ExecutionContext.cs]: Because is read only, and maybe we need it for debuggers or something like that

**Pull request merged**